### PR TITLE
CORE-913 / CORE-914: XRP and HBAR handlers

### PR DESCRIPTION
**Pull request by ewerx**

### CODE REVIEW

File: WalletKitCore/src/crypto/BRCryptoTransfer.c [hunk 1]
```diff
@@ -123,13 +129,14 @@ cryptoTransferGetTargetAddress (BRCryptoTransfer transfer) {
...
+    return NULL == transfer->amount ? NULL : cryptoAmountCreate (cryptoAmountGetUnit(transfer->amount),
+                                                                 isNegative,
+                                                                 cryptoAmountGetValue(transfer->amount));
 }
 
-
 extern BRCryptoAmount
 cryptoTransferGetAmount (BRCryptoTransfer transfer) {
-    return cryptoTransferGetAmountAsSign (transfer, CRYPTO_FALSE);
+    return NULL == transfer->amount ? NULL : cryptoAmountTake (transfer->amount);
```
Comment (EBGToo) [inline WalletKitCore/src/crypto/BRCryptoTransfer.c:139]: The `cryptoAmountTake()` (and `cryptoAmountGive()`) now handle a NULL argument - by simply returning it.  Thus, cases like the above can be replaced with `return cryptoAmountTake(...)`

File: WalletKitCore/src/crypto/BRCryptoTransferP.h [hunk 2]
```diff
@@ -96,6 +87,10 @@ struct BRCryptoTransferRecord {
 
     /// Actually this can be derived from { btc.fee / txSize(btc.tid), txSize(btc.tid) }
     BRCryptoFeeBasis feeBasisEstimated;
+    
+    BRCryptoTransferDirection direction;
+    
+    BRCryptoAmount amount;
```
Comment (EBGToo) [inline WalletKitCore/src/crypto/BRCryptoTransferP.h:93]: Maybe a comment here to indicated if this value is 'signed' or not.

File: WalletKitCore/src/crypto/handlers/hbar/BRCryptoSupportHBAR.c [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+//  BRCryptoSupportHBAR.c
+//
+//
+//  Created by Ehsan Rezaie on 2020-05-19.
+//
+
+#include "BRCryptoHBAR.h"
+#include "hedera/BRHederaBase.h"
+#include "../../BRCryptoHashP.h"
+#include "../../BRCryptoAmountP.h"
```
Comment (EBGToo) [inline WalletKitCore/src/crypto/handlers/hbar/BRCryptoSupportHBAR.c:11]: Can you replace `../../` with `crypto/` and in other files as well when accessing `BRCrypto*P.h`.  Need a general scrub of this... (and adding header/copyright comments as well - but that can be later.)

File: WalletKitCore/src/crypto/BRCryptoTransfer.c (hunk 1 above)
Comment (EBGToo) [reply WalletKitCore/src/crypto/BRCryptoTransfer.c:139]: And, just to note, in 'release' functions `if (NULL != obj) cryptoObjGive(obj)` becomes `cryptoObjGive(obj)`.

Comment (ewerx) [reply WalletKitCore/src/crypto/BRCryptoTransfer.c]: updated

File: WalletKitCore/src/crypto/BRCryptoTransferP.h (hunk 2 above)
Comment (ewerx) [reply WalletKitCore/src/crypto/BRCryptoTransferP.h:94]: added

File: WalletKitCore/src/crypto/handlers/hbar/BRCryptoSupportHBAR.c (hunk 3 above)
Comment (ewerx) [reply WalletKitCore/src/crypto/handlers/hbar/BRCryptoSupportHBAR.c]: fixed

**Pull request merged**